### PR TITLE
move build configuration to `pyproject.toml`

### PR DESCRIPTION
**Pull request by zacharyburnett**
`setuptools` now supports the `[project]` table, which is defined by [PEP621](https://peps.python.org/pep-0621).

Additionally, `setuptools` now supports its own entry in `pyproject.toml` called `[tool.setuptools]` (https://github.com/pypa/setuptools/issues/1688, https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration); however, it comes with the following disclaimer:
> Support for declaring configurations not standardized by [PEP 621](https://peps.python.org/pep-0621/) (i.e. the `[tool.setuptools]` table), is still in **beta** stage and might change in future releases.

Reading `toml` is supported natively in Python 3.11 with `tomllib`

Given this, we can attempt to consolidate the build configuration into a single `pyproject.toml` file that can possibly be read by other build systems in the future.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -10,16 +10,21 @@
 #
 # All configuration values have a default; values that are commented out
 # serve to show the default.
+from datetime import datetime
+from pathlib import Path
 
-import sys, os
+import tomli
```
Comment (zacharyburnett) [inline docs/source/conf.py]:
```suggestion
if sys.version_info < (3, 11):
     import tomli as tomllib
 else:
     import tomllib
```

File: docs/source/conf.py [hunk 2]
```diff
@@ -10,16 +10,21 @@
...
 
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
-#sys.path.append(os.path.abspath('.'))
+# sys.path.append(os.path.abspath('.'))
 
 # -- General configuration -----------------------------------------------------
+with open(Path(__file__).parent.parent.parent / "pyproject.toml", "rb") as configuration_file:
+    conf = tomli.load(configuration_file)
```
Comment (zacharyburnett) [inline docs/source/conf.py]:
```suggestion
    conf = tomllib.load(configuration_file)
```

File: pyproject.toml [hunk 3]
```diff
@@ -1,10 +1,57 @@
+[project]
+name = "stsci.image"
+description = "Image array manipulation functions"
+readme = "README.md"
+requires-python = ">=3.6"
```
Comment (nden) [inline pyproject.toml]:
```suggestion
requires-python = ">=3.9"
```

File: pyproject.toml [hunk 4]
```diff
@@ -1,10 +1,57 @@
...
+classifiers = [
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Topic :: Scientific/Engineering :: Astronomy",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+dependencies = [
+    "numpy>=1.14",
```
Comment (nden) [inline pyproject.toml]:
```suggestion
    "numpy",
```

File: pyproject.toml [hunk 5]
```diff
@@ -1,10 +1,57 @@
...
 [build-system]
 requires = [
-    "setuptools>=42",
-    "setuptools_scm",
+    "setuptools >=61.2",
+    "setuptools_scm[toml] >=6.2",
     "wheel",
     "oldest-supported-numpy; python_version>='3.7'",
-    ]
+    "numpy==1.14.6; python_version<'3.7'",
```
Comment (nden) [inline pyproject.toml]: I odn't think we need to support these. This package is used only by drizzlepac. Drizzlepac requires python>=3.9 beause it needs astropy. Anything older than this is not supported by astropy.

**Review by nden — APPROVED** [review]
LGTM.
Just need to remove the python 3.8 test from ci.yaml.

**Review by mcara — APPROVED** [review]

**Comment by zacharyburnett** [comment]
thanks!

**Pull request merged**